project/apps/web/app/questions/components/CreateModal.tsx

```diff
@@ -0,0 +1,201 @@
+              render={({ field }) => (
+                <FormItem>
+                  <FormLabel className="text-black">Description</FormLabel>
+                  <FormControl>
+                    <Textarea
+                      className="resize-y max-h-64"
+                      placeholder="Enter description"
+                      spellCheck={false}
+                      {...field}
+                    />
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+
+            {/* Complexity Dropdown */}
+            <FormField
+              control={form.control}
+              name="q_complexity"
+              render={({ field }) => (
+                <FormItem>
+                  <FormLabel className="text-black">Complexity</FormLabel>
+                  <FormControl>
+                    <Select value={field.value} onValueChange={field.onChange}>
+                      <SelectTrigger>
+                        <SelectValue placeholder="Select complexity" />
+                      </SelectTrigger>
+                      <SelectContent>
+                        {Object.values(COMPLEXITY).map((level) => (
+                          <SelectItem key={level} value={level}>
+                            {level}
+                          </SelectItem>
+                        ))}
+                      </SelectContent>
+                    </Select>
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+
+            {/* Categories Multi-Select */}
+            <FormField
+              control={form.control}
+              name="q_category"
+              render={() => (
+                <FormItem>
+                  <FormLabel className="text-black">Categories</FormLabel>
+                  <FormControl>
+                    <div className="flex flex-wrap gap-2">
+                      {categories.map((cat) => (
+                        <Badge
+                          key={cat}
+                          onClick={() => toggleCategory(cat)}
+                          variant={
+                            selectedCategories.includes(cat)
+                              ? "default"
+                              : "secondary"
+                          }
+                          className="cursor-pointer"
+                        >
+                          {cat}
+                        </Badge>
+                      ))}
+                    </div>
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+
+            <DialogFooter>
+              <Button
+                type="button"
+                variant="outline"
+                onClick={() => setOpen(false)}
+              >
+                Cancel
+              </Button>
+              <Button type="submit" variant="default">
```

Comment:
nit: disable button if form has errors.
```suggestion
              <Button
                type="submit"
                variant="default"
                disabled={Object.keys(form.formState.errors).length !== 0}
              >
```